### PR TITLE
Remove inherent in pallet-babe

### PR DESCRIPTION
**Pull request by thiolliere**
part of https://github.com/paritytech/substrate/issues/8029

Remove pallet babe inherent check and move it to OnTimestampSet.

companion: https://github.com/paritytech/polkadot/pull/2438

Test has been changed to now include the pre-runtime digest for babe slot. Because it is now checked

## Breaking change

babe pallet must be set in pallet_timestamp::Config::OnTimestampSet in order to be called when timestamp is set.
the part `Inherent` of babe pallet must be removed from construct_runtime (as inherent is no longer implemented).

### CODE REVIEW

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -1012,7 +1012,7 @@ construct_runtime!(
 	{
 		System: frame_system::{Module, Call, Config, Storage, Event<T>},
 		Utility: pallet_utility::{Module, Call, Event},
-		Babe: pallet_babe::{Module, Call, Storage, Config, Inherent, ValidateUnsigned},
+		Babe: pallet_babe::{Module, Call, Storage, Config, ValidateUnsigned},
```
Comment (bkchr) [inline bin/node/runtime/src/lib.rs:1015]: You did not added `Babe` to the `OnTimestampSet` associated types :D

Comment (andresilva) [reply bin/node/runtime/src/lib.rs:1015]: I think it should already be there.

**Review by andresilva — APPROVED** [review]
lgtm. needs a polkadot companion though

File: bin/node/runtime/src/lib.rs (hunk 1 above)
Comment (thiolliere) [reply bin/node/runtime/src/lib.rs:1015]: yes somehow Babe was in `OnTimestampSet` for all runtimes (same in polkadot companion)

**Comment by thiolliere** [comment]
yes companion now here: https://github.com/paritytech/polkadot/pull/2438

File: bin/node/runtime/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply bin/node/runtime/src/lib.rs:1015]: Ahh :D

**Review by bkchr — APPROVED** [review]

**Pull request merged**